src/util/uint.rs

```diff
@@ -96,22 +98,22 @@ macro_rules! construct_uint {
             }
 
             fn _from_be_slice(bytes: &[u8]) -> $name {
-                use super::endian::slice_to_u64_be;
                 let mut slice = [0u64; $n_words];
                 slice.iter_mut()
                     .rev()
                     .zip(bytes.chunks(8))
-                    .for_each(|(word, bytes)| *word = slice_to_u64_be(bytes));
+                    .for_each(|(word, bytes)| {
+                        *word = u64::from_be_bytes(bytes.try_into().expect("8 byte chunk"));
+                    });
                 $name(slice)
             }
 
             /// Convert a big integer into a byte array using big-endian encoding
             pub fn to_be_bytes(self) -> [u8; $n_words * 8] {
-                use super::endian::u64_to_array_be;
                 let mut res = [0; $n_words * 8];
                 for i in 0..$n_words {
                     let start = i * 8;
-                    res[start..start+8].copy_from_slice(&u64_to_array_be(self.0[$n_words - (i+1)]));
+                    res[start..start+8].copy_from_slice(&self.0[$n_words - (i+1)].to_be_bytes());
                 }
```

Comment:
Looks like this whole loop could use `chunks_exact_mut()` instead of manual indexing.

Reply:
Will use `chunks_exact`, perhaps I'll do that as a separate PR before this one.